Assets/Scripts/Ammunition.cs

```diff
@@ -29,8 +30,32 @@ public void PickUp()
             body.detectCollisions = false;
             body.isKinematic = true;
 
-            var size = GetComponent<MeshFilter>().mesh.bounds.size;
-            transform.localScale = new Vector3(inHandSize.x / size.x, inHandSize.y / size.y, inHandSize.z / size.z);
+            var meshRenderers = new List<MeshRenderer>();
+            GetComponentsInChildren(meshRenderers);
+
+            var meshRendererCount = meshRenderers.Count;
+            if (meshRendererCount > 0)
+            {
+                var bounds = meshRenderers[0].bounds;
+                for (var i = 1; i < meshRendererCount; ++i)
+                {
+                    bounds.Encapsulate(meshRenderers[i].bounds);
+                }
+
+                var size = bounds.size;
+                var handSizeRatio = new Vector3(
+                    inHandSize.x / size.x,
+                    inHandSize.y / size.y,
+                    inHandSize.z / size.z
+                );
+
+                transform.localScale = Vector3.Scale(transform.localScale, handSizeRatio);
+            }
+            else
+            {
+                Debug.LogWarning("Was unable to resize object to fit hand; MeshRenderer is missing.");
```

Comment:
nit: Recommended to add the context in Unity's log messages. They can map to GameObjects or components, so when you click the log message it highlights the target GameObject.

```suggestion
                Debug.LogWarning("Was unable to resize object to fit hand; MeshRenderer is missing.", this);
```